GSoC-2025/ethology: Low-shot detection for semi-automatic labelling (Rajat Kriplani).md

```diff
@@ -0,0 +1,108 @@
+    *   Export the final set of annotations (manual + reviewed/corrected) in a format compatible with `ethology` (e.g., save to COCO JSON or directly as a DataFrame).
+*   Unit tests covering the core backend functionality and basic widget interactions.
+*   User documentation explaining the workflow, usage of the Napari widget, and any model-specific considerations.
+
+**Detailed Weekly Timeline:**
+
+| Week | Dates (Approx. GSoC 2025) | Planned Tasks                                                                                                                                                              |
+| :--- | :------------------------ | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **CB** | (Community Bonding)       | Deep dive into GeCo & CountGD papers/code. Set up dev environment thoroughly. Experiment with Napari widget basics. Draft detailed API & preliminary widget UI mockups. Discuss designs with mentors. |
+| 1    | Late May                  | Finalize API design for backend inference. Set up backend module structure in `ethology`. Implement basic image/annotation loading for backend function. Start Napari widget boilerplate (loading images). |
+| 2    | Late May/Early June       | Implement core inference logic for the first model (e.g., GeCo) in the backend, focusing on processing input examples and running the model.                               |
+| 3    | Early June                | Refine backend inference output to standardized DataFrame format. Add backend unit tests. Implement basic bounding box drawing/selection tool in Napari widget.        |
+| 4    | Mid June                  | Integrate backend inference call into Napari widget. Display model output boxes on Napari canvas.                                                                        |
+| 5    | Mid/Late June             | Implement review/correction features in Napari: selecting, deleting model-generated boxes.                                                                               |
+| 6    | Late June/Early July      | Implement review/correction features in Napari: adjusting/editing model boxes, adding new manual boxes. **(Midterm: Core workflow demonstrable)**                         |
+| 7    | Early July                | Refactor backend API for potential second model integration. Implement saving/exporting functionality for the final annotations from Napari.                             |
+| 8    | Mid July                  | *If pursuing stretch goal:* Integrate second model (e.g., CountGD) OR *If sticking to minimal:* Add more robust testing for backend and frontend interactions.           |
+| 9    | Mid/Late July             | Write user documentation (workflow guide, widget usage).                                                                                                                 |
```

Comment:
We usually write tests and docs as we implement each feature, rather than all at the end. I would recommend accounting for the time for docs and tests along each feature.